src/app/components/core/configuration-manager/configuration-manager.component.ts

```diff
@@ -139,56 +102,22 @@ export class ConfigurationManagerComponent implements OnInit {
         });
   }
 
-  public onNodeToggleExpanded(event) {
-    event.node.data.children = [];
-    if (event.node.isExpanded) {
-      this.configService.getChildren(event.node.data.id).
-        subscribe(
-          (data) => {
-            data['categories'].forEach(element => {
-              if (element.hasOwnProperty('displayName')) {
-                event.node.data.children.push({
-                  id: element.key,
-                  name: element.displayName,
-                  description: element.description,
-                  hasChildren: true,
-                  children: []
-                });
-              } else {
-                event.node.data.children.push({
-                  id: element.key,
-                  name: element.description,
-                  description: element.description,
-                  hasChildren: true,
-                  children: []
-                });
-              }
-              this.tree.treeModel.update();
-            });
-          }, error => {
-            if (error.status === 0) {
-              console.log('service down ', error);
-            } else {
-              this.alertService.error(error.statusText);
-            }
-          });
-    }
+  addNameKey(childrenData) {
```

Comment:
What is addNameKey?

Reply:
`tree-root` component works on the basis of `name` key. And we don't have `name` key in the object received from API. So I've added it.